x/evm/types/state_object.go

```diff
@@ -142,6 +142,9 @@ func (so *stateObject) setState(key, value ethcmn.Hash) {
 	so.dirtyStorage = append(so.dirtyStorage, NewState(key, value))
 	idx = len(so.dirtyStorage) - 1
 	so.keyToDirtyStorageIndex[key] = idx
+	idx = len(so.originStorage)
```

Comment:
@noot should this be:
```suggestion
	idx = len(so.originStorage) - 1
```

Reply:
moved `append` line above this, then changed to `idx = len(so.originStorage) - 1`